students/Michael Hrstka/Lab 1/RGB_LED_Design_Challenge/RGB_LED_Design_Challenge.ino

```diff
@@ -0,0 +1,146 @@
+int ledR = 3;
+int ledG = 5;
+int ledB = 6;
+int btn = 12;
+int state = 0;
+int numState = 4;
+
+void setup() {
+  /* Setup pins */
+  pinMode(ledR, OUTPUT);
+  pinMode(ledG, OUTPUT);
+  pinMode(ledB, OUTPUT);
+  pinMode(btn, INPUT);
+}
+
+void chkBtn() {
+  if (digitalRead(btn) == HIGH) {
+    state = state >= numState ? 0 : state + 1;
+    while (digitalRead(btn) == HIGH){ // Debounce! (Wait for release)
+      delay(1);
+    }
+  }
+}
+
+/* Rapid white LED flash */
+bool ledBlink = true;
+void whiteLED() {
+  digitalWrite(ledR, ledBlink);
+  digitalWrite(ledG, ledBlink);
+  digitalWrite(ledB, ledBlink);
+
+  ledBlink = !ledBlink; // BLINK BLINK BLINK!!!
+  delay(100);
```

Comment:
delay? delay?!?

Reply:
Delay!!! (Existing artifact from before discussing why delay is bad, never broke anything so I never fixed it)